data_steward/cdr_cleaner/reporter.py

```diff
@@ -0,0 +1,306 @@
+
+    :returns: a list of dictionaries representing the requested fields.
+    """
+
+    report_rows = []
+    for rule in control.DATA_STAGE_RULES_MAPPING.get(data_stage, []):
+        rule_info = {}
+
+        try:
+            clazz = rule[0]
+            instance = clazz('foo', 'bar', 'baz')
+            LOGGER.info(f"{clazz} ducktyped to a class")
+
+        except (RuntimeError, TypeError, HttpError, BadRequest):
+            LOGGER.info(f"{rule} did NOT ducktype to a class")
+            rule_info = get_function_info(clazz, fields_list)
+            report_rows.append(rule_info)
+
+        else:
+            try:
+                # this is a class
+                _ = instance.get_query_specs()
+                LOGGER.info(f"{clazz} is a class")
+                for field in fields_list:
+                    try:
+                        value = 'NO DATA'
+                        if field in report.FIELDS_PROPERTIES_MAP:
+                            func = report.FIELDS_PROPERTIES_MAP[field]
+                            value = getattr(instance, func, 'no data')
+                        elif field in report.FIELDS_METHODS_MAP:
+                            func = report.FIELDS_METHODS_MAP[field]
+                            value = getattr(instance, func, 'no data')()
+                        elif field in report.CLASS_ATTRIBUTES_MAP:
+                            func = report.CLASS_ATTRIBUTES_MAP[field]
+
+                            value = None
+                            for item in func.split('.'):
+                                if not value:
+                                    value = getattr(instance, item)
+                                else:
+                                    value = getattr(value, item)
+
+                        rule_info[field] = value
+
+                    except AttributeError:
+                        # an error occurred trying to access an expected attribute.
+                        # did the base class definition change recently?
+                        LOGGER.exception(
+                            f'An error occurred trying to get the value for {field}'
+                        )
+                        rule_info[field] = 'unknown'
+            except (TypeError, AttributeError):
+                # an error occurred indicating this is not a rule extending the
+                # base cleaning rule.  provide the info we can and move on.
+                LOGGER.exception(f'{clazz} is not a class')
+                # this is a function
+                rule_info = get_function_info(clazz, fields_list)
+
+            report_rows.append(rule_info)
+
+    return report_rows
+
+
+def separate_sql_statements(rules_values):
+    """
+    Separate SQL statements into items with other identical fields.
+
+    This must maintain the SQL statement order.
+    """
+    separated_rules_values = []
+    for rule_values in rules_values:
+        sql_list = []
+        separated = dict()
```

Comment:
Is this assignment necessary since you are declaring it again below in 168 and 173?

Reply:
Going back to address this, i've found another bug.

Reply:
Bug addressed.  Good catch.